packages/rest/src/lib/REST.ts

```diff
@@ -233,6 +233,10 @@ export class REST extends EventEmitter {
 		this.on('removeListener', (name, listener) => {
 			if (name === RESTEvents.Request || name === RESTEvents.Response) this.requestManager.off(name, listener);
 		});
+
+		if (options.agent) {
+			setGlobalDispatcher(options.agent);
```

Comment:
Won't this cause issues? Should it be global? I'd argue the agent should only affect requests done through the module

Reply:
Setting is per request is possible with `undici.request` but causes issues with the tests because they use an Agent too... I'll get around to fixing this however.

Reply:
This is much easier to fix than I previously thought.